flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolRequestCompletionTest.java

```diff
@@ -129,22 +115,13 @@ private void runSlotRequestCompletionTest(
 		}
 	}
 
-	private SlotPoolImpl setUpSlotPoolAndConnectToResourceManager() throws Exception {
-		final SlotPoolImpl slotPool = setUpSlotPool();
-		connectToResourceManager(slotPool);
-
-		return slotPool;
-	}
-
 	private void connectToResourceManager(SlotPoolImpl slotPool) {
 		slotPool.connectToResourceManager(resourceManagerGateway);
 	}
 
-	private SlotPoolImpl setUpSlotPool() throws Exception {
-		final SlotPoolImpl slotPool = new TestingSlotPoolImpl(new JobID());
-
-		slotPool.start(JobMasterId.generate(), "foobar", ComponentMainThreadExecutorServiceAdapter.forMainThread());
-
+	private TestingSlotPoolImpl createAndSetUpSlotPoolWithoutResourceManager() throws Exception {
+		final TestingSlotPoolImpl slotPool = new TestingSlotPoolImpl(new JobID());
```

Comment:
can we use `SlotPoolBuilder` to do this by setting `resourceManagerGateway` to `null`?